src/utils/prepareSwapTransactions.test.ts

```diff
@@ -0,0 +1,288 @@
+import { ChainType, SquidCallType } from '@0xsquid/squid-types'
+import { NetworkId } from '../types/networkId'
+import { prepareSwapTransactions } from './prepareSwapTransactions'
+import { simulateTransactions } from '../runtime/simulateTransactions'
+import got from 'got'
+import { Address } from 'viem'
+
+const mockGotPostJson = jest.fn()
+const mockReadContract = jest.fn()
+
+jest.mock('got', () => ({
+  HTTPError: jest.requireActual('got').HTTPError,
+  post: jest.fn(() => ({
+    json: mockGotPostJson,
+  })),
+}))
```

Comment:
nit: I started using https://mswjs.io/ in this repo for network request mocks.
So tests are agnostic from the fetch client being used.

For now request mocks are all centralized in [test/server.ts](https://github.com/valora-inc/hooks/blob/main/test/server.ts), but I think we could probably refactor to override / provide custom responses per test.

We can address this separately though.

Reply:
Ah, this is just for unit tests right? Happy to make a follow up later to adjust this. The got mocking is  pretty messy.